developer/python-django-devtools-reference.rst

```diff
@@ -0,0 +1,244 @@
+
+You can get started with django by following `this tutorial
+<https://www.djangoproject.com/start/>`_ or`this one
+<https://pythonprogramming.net/web-development-tutorials/>`_.
+
+Django REST framework
+---------------------
+
+`Django REST framework <https://www.django-rest-framework.org>`_
+is a powerful and flexible toolkit for building Web APIs and it's used
+and trusted by internationally recognised companies including Mozilla,
+Red Hat, Heroku, and Eventbrite.
+
+Here are some reasons why you would want to use the REST framework:
+- Simplicity, flexibility, quality, and test coverage of source code.
+- Powerful serialization engine compatible with both ORM and non-ORM
+data sources.
+- Clean, simple, views for Resources, using Django's new class based
+views.
+- HTTP response handling, content type negotiation using HTTP Accept
+headers.
+- Publishing of metadata along with querysets.
+
+Thus, DRF allows the flexibility to extend and customize the framework’s
+tools according to programmer’s demands that greatly reduces development
+time. You can find some more details with examples and tutorials
+`on the official DRF website <https://www.django-rest-framework.org>`_
+or consider checking out the DRF tutorial along with some other tutorials
+on `this website <https://wsvincent.com/>`_.
+
+Django Extensions
+-----------------
+
+`Django Extensions <https://django-extensions.readthedocs.io/>`_ is a
+collection of custom extensions for the Django Framework. These include
+management commands, additional database fields, admin extensions and
+much more but we will focus more on 2 of them for now, namely
+``shell_plus`` and ``runserver_plus``.
+
+Django Extensions can be installed with:
+
+``pip install django-extensions``
+
+`shell_plus
+<django-extensions.readthedocs.io/en/latest/shell_plus.html>`_:
+Django shell with autoloading of the apps database models and subclasses
+of user-defined classes.
+
+`runserver_plus
+<django-extensions.readthedocs.io/en/latest/runserver_plus.html>`_:
+Typical runserver with Werkzeug debugger baked in.
+
+Django Debug Toolbar
+--------------------
+
+The `Django Debug Toolbar <https://django-debug-toolbar.readthedocs.io/>`_
+is a configurable set of panels that display various debug information
+about the current request/response and when clicked, display more details
+about the panel’s content.
+
+It can be installed with:
+
+``pip install django-debug-toolbar``
+
+More details can be found on its `website
+<https://django-debug-toolbar.readthedocs.io/en/latest/>`_
+or on `pypi <https://pypi.org/project/django-debug-toolbar/>`_.
+
+IPython and ipdb
+----------------
+
+`IPython <https://ipython.org>`_ (Interactive Python) is a command shell
+for interactive computing in multiple programming languages, originally
+developed for the Python programming language, that offers introspection,
+rich media, shell syntax, tab completion, and history.
+
+It provides:
+- A powerful interactive shell.
```

Comment:
I guess you meant this paragraph to be a bulleted list, but it is displayed as regular text. Maybe you need an empty line above.

Reply:
nevermind. I didn't know I could see the text displayed as it would normally be displayed before the PR is merged. Fixed all that.